### PR TITLE
Improve pipe handling on linux

### LINKED ISSUE #3068: Don't report broken pipe on early pager exit
### I did this

```
kanidm person list|less
```

Scrolled through a bit, but not all the way to the bottom - then hit `q`.

### I expected the following

No errors.

### This happened instead

```
thread 'main' panicked at library/std/src/io/stdio.rs:1118:9:
failed printing to stdout: Broken pipe (os error 32)
```

### Kanidm version details

- Output of `kanidm(d) version`: `kanidm 1.3.3`
- Are you running it in a container? If so, which image/tag?: n/a
- If not a container, how'd you install it: zypper
- Operating System / Version (On Unix please post the output of `uname -a`): openSUSE Leap 15.6

### Any other comments

This is only a cosmetic problem. There are workarounds such as piping into some other tool in between or just ignoring the message.
If fully consuming the pipe by scrolling all the way to the bottom in `less`, no errors are reported.
Also this is only observed in the default output format, not with `-o json`.

### PR DESCRIPTION
**Pull request by Firstyear**
# Change summary

- Handle sigpipe on linux to prevent needless backtraces

Fixes #3068

Checklist

- [ x ] This PR contains no AI generated code
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Review by yaleman — APPROVED** [review]

File: tools/cli/src/cli/main.rs [hunk 1]
```diff
@@ -13,13 +13,39 @@
...
+use std::process::ExitCode;
 use std::thread;
 use tokio::runtime;
+use tokio::signal::unix::{signal, SignalKind};
 use tracing_subscriber::filter::LevelFilter;
 use tracing_subscriber::prelude::*;
 use tracing_subscriber::{fmt, EnvFilter};
 
-fn main() {
+async fn signal_handler(opt: KanidmClientParser) -> ExitCode {
```
Comment (yaleman) [inline tools/cli/src/cli/main.rs:24]: Actually... this needs a Unix /windows target split

**Review by yaleman — CHANGES_REQUESTED** [review]

File: tools/cli/src/cli/main.rs (hunk 1 above)
Comment (Firstyear) [reply tools/cli/src/cli/main.rs:24]: Will do

**Review by yaleman — APPROVED** [review]
Yay! Thanks

**Comment by tacerus** [comment]
Cool stuff, thank you!

**Pull request merged**